data_steward/tools/copy_dataset_to_output_prod.py

```diff
@@ -0,0 +1,120 @@
+"""
+This script copies a dataset from a source project to the output prod project
+and runs the recreate_person tool.
+"""
+
+# Python imports
+import logging
+import argparse
+
+# Project imports
+from utils import bq, auth, pipeline_logging
+from tools.create_tier import (TIER_LIST, DEID_STAGE_LIST, get_dataset_name)
+from tools.recreate_person import update_person
+
+SCOPES = [
+    'https://www.googleapis.com/auth/bigquery',
+    'https://www.googleapis.com/auth/devstorage.read_write',
+    'https://www.googleapis.com/auth/cloud-platform'
+]
+LOGGER = logging.getLogger(__name__)
+
+
+def get_arg_parser() -> argparse.ArgumentParser:
+    """
+    Copy dataset from curation project to output-prod
+    """
+    argument_parser = argparse.ArgumentParser(description=__doc__)
+    argument_parser.add_argument(
+        '--run_as',
+        action='store',
+        dest='run_as_email',
+        help='Service account email address to impersonate',
+        required=True)
+    argument_parser.add_argument(
+        '-s',
+        '--src_project_id',
+        dest='src_project_id',
+        action='store',
+        help='Identifies the project containing the source dataset',
+        required=True)
+    argument_parser.add_argument('-o',
+                                 '--output_prod_project_id',
+                                 dest='output_prod_project_id',
+                                 action='store',
+                                 help='Identifies the output-prod project.',
+                                 required=True)
+    argument_parser.add_argument('-d',
+                                 '--src_dataset_id',
+                                 dest='src_dataset_id',
+                                 action='store',
+                                 help='The source dataset to copy.',
+                                 required=True)
+    argument_parser.add_argument(
+        '-r',
+        '--release_tag',
+        action='store',
+        dest='release_tag',
+        help='release tag for dataset in the format of YYYYq#r#',
```

Comment:
I believe the release tag for output-prod is different from the naming used in prod and should be all caps - this can be a check during parsing (check if isupper()) for that argument type https://docs.python.org/3/library/argparse.html#type.

Reply:
Looking at the `C2021Q2R1` as an example, it looks like the dataset name is all caps but the release tag label (`2021q2r1`) is lowercase. I'm thinking I can still accept the lowercase tag as an arg and then convert it to caps for the dataset name. Is that fine?

Reply:
I created a new `get_dataset_name` that parameterizes the name properly. It also removes "deid" from the dataset name.

Reply:
Not always. This means the user could type in the internal release tag by mistake (eg `2021q3r4`), instead of the output release tag (eg `2021Q3R1`). Output release number should be set to 1 unless there's a hotfix, regardless of the internal release number, so converting it to upper may cause issues.

Reply:
Ok I see, I'll change the arguments then.